cmd/lncli/commands.go

```diff
@@ -651,27 +656,30 @@ var closeChannelCommand = cli.Command{
 				"sat/byte that should be used when crafting " +
 				"the transaction",
 		},
+		cli.BoolFlag{
+			Name:  "all",
+			Usage: "close all existing active and inactive channels",
+		},
 	},
 	Action: actionDecorator(closeChannel),
 }
 
 func closeChannel(ctx *cli.Context) error {
-	ctxb := context.Background()
-	client, cleanUp := getClient(ctx)
-	defer cleanUp()
-
-	args := ctx.Args()
-	var (
-		txid string
-		err  error
-	)
-
-	// Show command help if no arguments provided
 	if ctx.NArg() == 0 && ctx.NumFlags() == 0 {
 		cli.ShowCommandHelp(ctx, "closechannel")
 		return nil
 	}
 
+	if ctx.Bool("all") {
+		// Make sure only the "all" flag was set.
+		if ctx.NArg() != 0 || ctx.NumFlags() != 1 {
+			return fmt.Errorf("--all cannot be used with other " +
+				"args/flags")
+		}
+
```

Comment:
If we could prompt the user here to confirm that they indeed want to close all channels, I think that would go a long way in preventing an apocalypse :)

Reply:
If we decide to create a `closeallchannels` command, we could mimic how `rm` works:

`closeallchannels` would ask for confirmation for every inactive channel only and attempt to close all channels.

`closeallchannels --inactive-only` would ask for confirmation for every inactive channel and attempt to close them.

`closeallchannels --force` would only ask for confirmation once and attempt to close all channels.

`closeallchannels --inactive-only --force` would ask for confirmation once and attempt to close all inactive channels only.

Reply:
Excellent suggestion, I like that our mass channel teardown logic would be done behind a separate rpc altogether. Mimicking `rm` makes sense from a UX perspective, and the flags you suggest work well 😊

Reply:
Is a separate RPC necessary though? The confirmation aspect would be a bit more tricky to handle when using anything other than `lncli`.